compiler/rustc_expand/src/expand.rs

```diff
@@ -1195,6 +1159,30 @@ macro_rules! assign_id {
 }
 
 impl<'a, 'b> MutVisitor for InvocationCollector<'a, 'b> {
+    fn visit_crate(&mut self, krate: &mut ast::Crate) {
+        let span = krate.span;
+        let empty_crate =
+            || ast::Crate { attrs: Vec::new(), items: Vec::new(), span, is_placeholder: None };
+        let mut fold_crate = |krate: ast::Crate| {
+            let mut krate = match self.configure(krate) {
+                Some(krate) => krate,
+                None => return empty_crate(),
```

Comment:
Can the entire crate really be removed with something like `#![cfg(FALSE)]`? Is there a test case for this?

Reply:
Yes, libcore has `#![cfg(not(test))]` in its root in particular.
There are some tests for this as well, e.g. `src\test\ui\issues\issue-34932.rs` and `src\test\ui\conditional-compilation\cfg-in-crate-1.rs`.